crates/rethnet_rpc_server/src/lib.rs

```diff
@@ -204,7 +211,7 @@ async fn handle_get_balance(
     block: Option<BlockSpec>,
 ) -> ResponseData<U256WithoutLeadingZeroes> {
     event!(Level::INFO, "eth_getBalance({address:?}, {block:?})");
-    match set_block_context(&state, block).await {
+    match set_block_context(&state, block.clone()).await {
```

Comment:
What made the clone necessary?

Reply:
I suspect that at some point I was doing some debug logging in here that required this clone, but clearly it's unnecessary at this point so i removed it in https://github.com/NomicFoundation/hardhat/pull/4095/commits/be61d53fe80b7aafb3484d84a8c4eb6631f87796

Reply:
Did clippy not raise an error about this?

Reply:
Nope, it ran cleanly both locally and in CI.